### ISSUE TITLE
Allow Returning Raw Data Instead of HTML

### ISSUE THREAD

**Issue by amxmln** [RafidMuhymin/astro-imagetools#150]
Hello there! I’m currently looking into using `astro-imagetools` to implement my own `<Img />` and `<Picture />` components in Vue.js.

This should be possible by using the API functions, but unfortunately the values are returned as HTML. I could parse those return values and get my data that way, but I think it would be nice to provide some additional API functions to just return the data itself, so a user can decide how they want to build the HTML from it.

Something like `getRawPicture(props)` that returns an array of sources, the placeholder image, image dimensions, etc.

**Comment by amxmln** [comment]
Yeah, that’s what I thought would be an option as well. What would you want to tweak about the function?

Totally get not having enough time to work on it—you‘ve done so much already!

I can’t commit to a PR right now, but I’ll look into it and update the issue accordingly.

**Comment by RafidMuhymin** [comment]
Happy to see your interest in the project.

> What would you want to tweak about the function?

Can't tell currently. Will have to check. I just said because it might need tweaking.
